tests/components/modbus/test_modbus_sensor.py

```diff
@@ -17,12 +19,42 @@
     DATA_TYPE_UINT,
 )
 from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import CONF_NAME
 
-from .conftest import run_test
+from .conftest import run_base_test
 
 _LOGGER = logging.getLogger(__name__)
 
 
+async def run_sensor_test(
+    hass, use_mock_hub, register_config, register_words, expected
+):
+    """Run test for sensor."""
+    sensor_name = "modbus_test_sensor"
+    entity_domain = SENSOR_DOMAIN
+    data_array = {
+        CONF_REGISTERS: [
+            dict(**{CONF_NAME: sensor_name, CONF_REGISTER: 1234}, **register_config)
+        ]
+    }
+
+    await run_base_test(
+        sensor_name,
+        hass,
+        use_mock_hub,
+        data_array,
+        register_config.get(CONF_REGISTER_TYPE),
+        entity_domain,
+        register_words,
+        expected,
+    )
+
+    # Check state
+    entity_id = f"{entity_domain}.{sensor_name}"
+    state = hass.states.get(entity_id).state
+    assert state == expected
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6243615)